### ISSUE TITLE
Got 'Object possibly undefined' when calling base class's optional method

### ISSUE THREAD

**Issue by feng-xiao** [Microsoft/TypeScript#14388]
When 'stickNullChecks' is turned on and calling base class's optional method with following code, got 'Object possibly undefined' error

**TypeScript Version:**  2.2.1

**Code**

_Sub-class:_

```ts
super.methodOptional && super.methodOptional(someAgs);

```
_Base-class:_

```ts
protected methodOptional?(someAgs): void;

```

**Expected behavior:**
No error

**Actual behavior:**
Got 'Object possibly undefined' error on _super.methodOptional_

**Comment by ahejlsberg** [comment]
Hmm, yes, we should probably track property access expressions of the form `super.xxx` in control flow analysis. We currently don't.

**Comment by sandersn** [comment]
Fix is up at #14956
